### PR TITLE
[Bugfix][HTTPS] Correct Handling of http Expect: 100-continue in

### PR DESCRIPTION
**Pull request by akhilkedia**
The previous code incorrectly handled a Expect: 100-continue response
to a POST request from the Server, causing the HTTP Parser
to think that the request was over.

HTTPS client's parser was incorrectly assuming the message was complete based on
the newline after 100-continue header. 

Added a Unit test case for this too at ./test/run_pass/test_https_post_status_codes.js.

IoT.js-DCO-1.0-Signed-off-by: Akhil Kedia akhil.kedia@samsung.com

### CODE REVIEW

**Comment by akhilkedia** [comment]
@sumin-lim I have uploaded the bugfix here.

File: test/run_pass/test_https_post_status_codes.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    "authorization": "Bearer 1718113118564ad495ad03f04116f379"
+  }
+};
+
+var getResponseHandler = function (res) {
+  var res_body = '';
+
+  assert.equal(200, res.statusCode);
+
+  var endHandler = function(){
```
Comment (LaszloLango) [inline test/run_pass/test_https_post_status_codes.js]: missing space before `{`

File: test/run_pass/test_https_post_status_codes.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    var response = JSON.parse(res_body);
+    assert.assert(response['data'], 'Recieved incorrect response from server');
+    isRequest1Finished = true;
+  };
+  res.on('end', endHandler);
+
+  res.on('data', function(chunk){
```
Comment (LaszloLango) [inline test/run_pass/test_https_post_status_codes.js]: ditto

File: test/run_pass/test_https_post_status_codes.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  "hostname": "api.artik.cloud",
+  "path": "/v1.1/messages",
+  "headers": {
+    "content-type": "application/json",
+    "content-length": data.length,
+    "authorization": "Bearer 1718113118564ad495ad03f04116f379"
+  }
+};
+
+var getResponseHandler = function (res) {
```
Comment (LaszloLango) [inline test/run_pass/test_https_post_status_codes.js]: No space needed: `function(res) {`

Comment (akhilkedia) [inline test/run_pass/test_https_post_status_codes.js]: Done.

File: test/run_pass/test_https_post_status_codes.js (hunk 1 above)
Comment (akhilkedia) [inline test/run_pass/test_https_post_status_codes.js]: Done.

File: test/run_pass/test_https_post_status_codes.js (hunk 2 above)
Comment (akhilkedia) [inline test/run_pass/test_https_post_status_codes.js]: Done.

**Comment by akhilkedia** [comment]
@LaszloLango  For future reference, is there any tool which you use for these syntax issues, or is it done by manual inspection?
The included `tools/checktidy.py` works wonders when it comes to C code, but misses several issues with JS syntax (such as the ones above.)

**Comment by LaszloLango** [comment]
@akhilkedia unfortunately not. There were some experiments with ESLint but the PR was dropped. So only manual inspection at the moment.

**Comment by yichoi** [comment]
LGTM

**Pull request merged**